Board.java

```diff
@@ -30,33 +28,123 @@ public Board playerMakesMove(char player, int position) {
         return this;
     }
 
+    public String getWinner() {
+        for (String combination : allWinningCombinations()) {
+            if (!combination.contains("x") && !combination.contains("-")) {
+                return "o";
+            }
+            if (!combination.contains("o") && !combination.contains("-")) {
+                return "x";
+            }
+        }
+        return null;
+    }
+
+    public boolean hasWinner() {
+        return getWinner() != null;
+    }
+
+
+    public boolean isGameOver() {
+        return hasWinner() || checkForDraw();
+    }
+
+    public ArrayList<String> allWinningCombinations() {
+        ArrayList<String> combinations = new ArrayList<>();
```

Comment:
What are these 'combinations' if not 'lines'?  You'll probably find Board class is growing quite long.. If it's getting up to 200 lines, you could think about extracting this code out into a Lines class...